gobblin-runtime/src/main/java/org/apache/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -291,6 +296,14 @@ protected void runWorkUnits(List<WorkUnit> workUnits) throws Exception {
       this.eventSubmitter.submit(countEventBuilder);
       LOG.info("Emitting WorkUnitsCreated Count: " + countEventBuilder.getCount());
 
+      MetricContext
+          metricContext = Instrumented.getMetricContext(ConfigUtils.configToState(ConfigFactory.empty()), getClass());
+      String workUnitCountName = MetricRegistry.name(ServiceMetricNames.GOBBLIN_SERVICE_PREFIX,
+          jobState.getProp(ConfigurationKeys.FLOW_GROUP_KEY), jobState.getProp(ConfigurationKeys.FLOW_NAME_KEY),
+          JobEvent.WORK_UNITS_CREATED);
+      ContextAwareGauge guage = metricContext.newContextAwareGauge(workUnitCountName, () -> workUnits.size());
```

Comment:
Where do we emit other metrics for GaaS-triggered jobs - inside the job itself or in the GaaS code?

I see that we are emitting CountEvent for work unit size earlier in the code. I assume it will land in GaaS and be available for processing.

Reply:
The issue here is that we don't have access to the number of work units easily from the GaaS code. One option will be to use the metadata in the CountEvent like you are suggesting and then emit actual metrics from the point its landing in GaaS (the KafkaAvroJobStatusMonitor) but that seemed like a hackier way of doing it.